src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -1070,23 +1110,74 @@ DatabaseShardImp::initConfig(std::lock_guard<std::mutex>&)
     }
 
     using namespace boost::filesystem;
-    if (!get_if_exists<path>(section, "path", dir_))
-        return fail("'path' missing");
+    if (!get_if_exists<path>(section, "main_storage_path", dir_))
+        return fail("'main_storage_path' missing");
 
     {
-        std::uint64_t sz;
-        if (!get_if_exists<std::uint64_t>(section, "max_size_gb", sz))
-            return fail("'max_size_gb' missing");
+        std::uint64_t count = 0;
+        get_if_exists<std::uint64_t>(section, "max_historical_shards", count);
+
+        maxHistoricalShards_ = count;
+
+        Section const& historicalShardPaths =
+            config.section(SECTION_HISTORICAL_SHARD_PATHS);
 
-        if ((sz << 30) < sz)
-            return fail("'max_size_gb' overflow");
+        auto values = historicalShardPaths.values();
 
-        // Minimum storage space required (in gigabytes)
-        if (sz < 10)
-            return fail("'max_size_gb' must be at least 10");
+        std::sort(values.begin(), values.end());
+        values.erase(std::unique(values.begin(), values.end()), values.end());
 
-        // Convert to bytes
-        maxFileSz_ = sz << 30;
+        std::string hPathError = "";
+
+        if (std::find_if(
```

Comment:
I'd rewrite this so it doesn't use `find_if`. Rational: 1) we aren't finding anything, so it's easy to misread the code, 2) Normally the lambdas to `find_if` are pure, here the lambda is creating directories 3) Even if the lambda were pure, an explicit range for loop is something most developers are used to seeing for code like this. Unless there's a compelling reason, it's good to use idioms other developers expect.

Reply:
Done